src/string.cr

```diff
@@ -900,7 +900,7 @@ class String
     end
   end
 
-  def unsafe_byte_at(index)
+  protected def unsafe_byte_at(index)
```

Comment:
When all calls to this method get replaced by `to_unsafe[index]` it could as well be removed entirely.

Reply:
I think we should really remove this method.

```crystal
string.to_unsafe[index]
string.unsafe_byte_at_index(index)
```

the second is even longer!

Also, I'd like to come back with the idea of having general benchmarks of the whole standard library. Then we could see which places don't need to be "unsafe" and work just fine with safe code.